library/Fission/Web/Heroku.hs

```diff
@@ -1,152 +1,29 @@
-  , Has Web.Host        cfg
-  , HasLogFunc          cfg
+  ( MonadDB          m
+  , MonadLocalIPFS   m
+  , MonadRemoteIPFS  m
+  , MonadThrow       m
+  , MonadLogger      m
+  , MonadTime        m
+  , MonadReader  cfg m
+  , Has Web.Host cfg
   )
-  => RIOServer cfg ProvisionAPI
-provision Request {uuid, region} = do
-  Web.Host url' <- Config.get
-  ipfsPeers     <- getExternalAddress >>= \case
-                     Right peers' ->
-                       pure peers'
-
-                     Left err -> do
-                       logError <| displayShow err
-                       return []
-
-  username <- liftIO <| User.genID
-  secret   <- liftIO <| Random.text 200
-
-  User.createWithHeroku uuid region username secret >>= \case
-    Left err ->
-      Web.Err.throw err
-
-    Right userID -> do
-      logInfo <| mconcat
-        [ "Provisioned UUID: "
-        , displayShow uuid
-        , " as "
-        , displayShow userID
-        ]
-
-      return Provision
-        { id      = userID
-        , peers   = ipfsPeers
-        , message = "Successfully provisioned Interplanetary Fission!"
-        , config  = User.Provision
-           { username = User.hashID userID
-           , password = Secret secret
-           , url      = url'
-           }
-        }
-
-type DeprovisionAPI = Capture "addon_id" UUID
-                   :> DeleteNoContent '[Heroku.MIME.VendorJSONv3] NoContent
-
-deprovision ::
-  ( MonadSelda      (RIO cfg)
-  , MonadMask       (RIO cfg)
-  , MonadRemoteIPFS (RIO cfg)
-  , HasLogFunc           cfg
-  )
-  => RIOServer cfg DeprovisionAPI
-deprovision uuid' = do
-  let err = Web.Err.ensureMaybe err410 -- HTTP 410 is specified by the Heroku AddOn docs
-
-  AddOn { addOnID } <- err =<< Query.oneEq Table.addOns #uuid uuid'
-  User  { userID }  <- err =<< Query.findOne do
-    user <- select Table.users
-    restrict <| user ! #herokuAddOnId .== literal (Just addOnID)
-            .&& user ! #active        .== true
-    return user
-
-  usersCIDs <- query do
-    uCIDs <- select Table.userCIDs
-    restrict (uCIDs ! #userFK .== literal userID)
-    return (uCIDs ! #cid)
-
-  cidOccur <- query do
-    (liveCID' :*: occurences') <- aggregate do
-      uCIDs <- select Table.userCIDs
-      theCID <- groupBy (uCIDs ! #cid)
-      return (theCID :*: count (uCIDs ! #cid))
-
-    restrict (liveCID' `isIn` fmap literal usersCIDs)
-    return <| liveCID' :*: occurences'
-
-  transaction do
-    deleteFrom_ Table.userCIDs <| #userFK `is` userID
-    deleteFrom_ Table.users    <| #userID `is` userID
-    deleteFrom_ Table.addOns   <| #uuid   `is` uuid'
-
-  let toUnpin = CID . Selda.first <$> filter ((== 1) . Selda.second) cidOccur
-  forM_ toUnpin <| IPFS.Pin.rm >=> \case
-    Left ipfsMsg -> do
-      logError <| "Unable to unpin CID: " <> display ipfsMsg
-      return ()
-
-    Right _ ->
-      return ()
-
-  return NoContent
+  => ServerT API m
+server = Provision.create
+    :<|> Deprovision.destroy
```

Comment:
Split these into their own files. They're the beefiest handlers in the project, and this file was getting long. Plus, it could be good to break out some handlers and whatnot if they continue to get more complex.